tools/private_domains_checker/PSLPrivateDomainsProcessor.py

```diff
@@ -0,0 +1,206 @@
+import datetime
+import json
+import os
+import pandas as pd
+import requests
+import whois
+
+
+def check_dns_status(domain):
+    def make_request():
+        try:
+            url = f"https://dns.google/resolve?name={domain}&type=NS"
```

Comment:
Google is a good source for this, but I do want to mention that this places authority on to Google - they could theoretically intervene in the resolution process on a given record.

Would we perhaps want to use an array of such resolvers that are randomly selected from?

Reply:
**Updated** - Good point about relying solely on Google for DNS, as it centralizes authority and potential intervention risks. Sometimes, it gives errors due to random reasons such as network fluctuations, so perhaps randomly selecting from multiple resolvers is still not error-proof. So, I decided to query two resolvers (Google and Cloudflare) and only take the value if they are consistent across the two: the updated implementation uses both Google and Cloudflare DNS resolvers. It queries both resolvers and only considers the result usable and consistent if both return the same status. If the results are inconsistent or if there are errors, it re-retries up to 5 times.